lib/base-class.js

```diff
@@ -0,0 +1,104 @@
+
+'use strct'
+
+const events = require('events')
+const os = require('os')
+const { spawn } = require('child_process')
+const path = require('path')
+const joinTrace = require('node-trace-log-join')
+const getLoggingPaths = require('./get-logging-paths')
+
+class Tool extends events.EventEmitter {
+  constructor ({
+    name,
+    customEnvHook
+  }) {
+    super()
+    this.name = name
+    this.customEnvHook = customEnvHook
```

Comment:
I expect users of this would add this method to the class (and use extend) themselves. So we should provide a base implementation (noop) in this class, and let the tools override this.

Reply:
I went for validating it and I added tests for it as well.